src/deadline/job_attachments/api/attachment.py

```diff
@@ -179,7 +182,11 @@ def _process_path_mapping(
                 f"Specified path mapping file {path_mapping_rules} is not valid."
             )
         with open(path_mapping_rules, encoding="utf8") as f:
-            path_mapping_rule_list.extend([PathMappingRule(**mapping) for mapping in json.load(f)])
+            data = json.load(f)
+            if "path_mapping_rules" in data:
+                data = data["path_mapping_rules"]
```

Comment:
Is it guaranteed that the data is a list of `PathMappingRule` if the JSON doesn't have a `path_mapping_rules` attribute? In what cases can this happen that we support?

Reply:
For openjd generated path mapping rule, it's guaranteed to be a list as long as the file exist.

https://github.com/OpenJobDescription/openjd-specifications/wiki/How-Jobs-Are-Run#path-mapping

Reply:
I'm still thinking about whether we should support if the JSON doesn't have a path_mapping_rules, i.e. non openjd format, keeping it here for now for backward compatibility. Either way, what format would be supported should be documented.

Reply:
Sounds good, if the behaviour accepted a list directly before, I think it's fine to keep it here.

For that case, I was considering whether we could do an extra check to at least make sure `data` is actually a `list` of `dict`, but it's probably not worth the code if the error message is clear enough and would be similar anyway (`TypeError`)

Reply:
Currently the error would be from L189 below as `TypeError: deadline.job_attachments.models.PathMappingRule() argument after ** must be a mapping, not str`. That can be a bit confusing, I can add some checks to provide explicit error message in an upcoming PR I'm currently working on.